pkg/mods/file/file.go

```diff
@@ -1,23 +1,75 @@
 package file
 
 import (
+	"errors"
 	"math/big"
 	"os"
 	"strconv"
 
 	"src.elv.sh/pkg/eval"
 	"src.elv.sh/pkg/eval/errs"
 	"src.elv.sh/pkg/eval/vals"
+	"src.elv.sh/pkg/sys"
 )
 
 var Ns = eval.BuildNsNamed("file").
 	AddGoFns(map[string]any{
 		"close":    close,
+		"is-tty":   isTty,
 		"open":     open,
 		"pipe":     pipe,
 		"truncate": truncate,
 	}).Ns()
 
+var errNotATTY = errors.New("not a tty")
+
+//elvdoc:fn is-tty
+//
+// ```elvish
+// file:is-tty $file-obj?
+// ```
+//
+// Does nothing if `$file-obj` is open on a tty (i.e., a terminal device); otherwise, raises the
+// "not a tty" exception. If the `$file-obj` argument is omitted the default output byte stream is
+// tested.
```

Comment:
The name of the function (`is-tty`) implies that it's a predicate. Like other predicates in Elvish, it should output a single boolean value instead of raising an exception.

Reply:
/face-palm Yes, of course. I have no idea why I decided it made more sense to raise an error.